bench/src/bin/bulk.rs

```diff
@@ -58,8 +59,11 @@ async fn server(mut incoming: quinn::Incoming, opt: Opt) -> Result<()> {
         };
         trace!("stream established");
 
-        let _: tokio::task::JoinHandle<Result<usize>> =
-            tokio::spawn(async move { drain_stream(&mut stream, opt.read_unordered).await });
+        let _: tokio::task::JoinHandle<Result<()>> = tokio::spawn(async move {
+            drain_stream(&mut recv_stream, opt.read_unordered).await?;
+            send_data_on_stream(&mut send_stream, opt.download_size).await?;
```

Comment:
Should these be concurrent? Sequential seems of limited value, since you could get similar data from just running two tests. Alternatively, use unidirectional streams in both directions, with handlers spawned independently?

Reply:
I thought about it. But that would require adding either a `join!` which adds overhead due to polling non-ready tasks, or adding more subtasks or futures-unordered. In the end I decided its not worth it since testing both directions at the  same time doesn't seem super useful, and sequential still gives useful values since at least the per-stream metrics only cover the actual transfer time inside the stream.

Reply:
If we use unidirectional streams in each direction, I don't think the actual number of subtasks needs to change, since streams for an unused transfer direction can be dynamically omitted.

That said, I don't see a clear requirement for the capability, so :shrug:, nonblocking.